backend/icpy/api/endpoints/media.py

```
@@ -50,15 +52,74 @@ async def get_image(
         
-        # Check if file exists
+        # Cache-first fast path
+        if not thumbnail:
+            try:
+                if cache and cache.has(image_id):
+                    b64 = cache.get(image_id)
+                    if b64:
+                        logger.info(f"[Media API] Cache hit for {image_id}; serving from memory")
+                        try:
+                            content = base64.b64decode(b64)
+                        except Exception:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)